### PR TITLE
Add the modulo operator

### PR DESCRIPTION
**Pull request by pateljm**
Added the modulo (`%`)operator, so we can now write queries like: 
```sql
SELECT i%2 FROM generate_series(1, 10) as GS(i);
```

### CODE REVIEW

File: expressions/scalar/tests/Scalar_unittest.cpp [hunk 1]
```diff
@@ -373,6 +373,7 @@ TEST_F(ScalarTest, ScalarBinaryExpressionSerializationTest) {
   checkScalarBinaryExpressionSerialization(BinaryOperationFactory::GetBinaryOperation(BinaryOperationID::kSubtract));
   checkScalarBinaryExpressionSerialization(BinaryOperationFactory::GetBinaryOperation(BinaryOperationID::kMultiply));
   checkScalarBinaryExpressionSerialization(BinaryOperationFactory::GetBinaryOperation(BinaryOperationID::kDivide));
+  checkScalarBinaryExpressionSerialization(BinaryOperationFactory::GetBinaryOperation(BinaryOperationID::kModulo));
```
Comment (zuyu) [inline expressions/scalar/tests/Scalar_unittest.cpp]: Update the copyright for this file as well.

**Comment by zuyu** [comment]
Please update the copyright for all the touched files, if they have not been modified since `2016`.

**Comment by jianqiao** [comment]
LGTM.

**Comment by cramja** [comment]
LGTM, merging.

**Pull request merged**